src/App/Acl.php

```diff
@@ -13,27 +13,73 @@
 
 use Ushahidi\Core\Tool\Permissions\Acl as AclInterface;
 use Ushahidi\Core\Entity\User;
+use Ushahidi\Core\Entity\Permission;
 use Ushahidi\Core\Entity\RoleRepository;
 
 class Acl implements AclInterface
 {
 	protected $role_repo;
+	protected $roles_enabled = false;
+	const DefaultRoles = [
+		'user'  => [Permission::EditOwnPosts]
+	];
 
 	public function setRoleRepo(RoleRepository $role_repo)
 	{
 		$this->role_repo = $role_repo;
 	}
 
+	public function setRolesEnabled($roles_enabled)
+	{
+		$this->roles_enabled = $roles_enabled;
+	}
+
+	/**
+	 * Check if custom roles are enabled for this deployment
+	 * @return boolean
+	 */
+	protected function hasRolesEnabled()
+	{
+		return (bool) $this->roles_enabled;
+	}
+
 	// Acl interface
 	public function hasPermission(User $user, $permission)
 	{
+		// If the user has no role, they have no permissions
 		if (!$user->role) {
 			return false;
 		}
 
+		// Don't check for permissions if we don't have the
+		// roles feature enabled
+		if ($this->hasRolesEnabled()) {
```

Comment:
So I didn't go as far as splitting the Acl in 2. But I've split the custom roles/default permissions into their method, and moved the handling of enabling/disabling into the Acl itself.